client/rpc/src/lib.rs

```diff
@@ -334,8 +347,24 @@ where
             .get_metadata(&file_key)
             .map_err(into_rpc_error)?
         {
-            None => return Ok(false),
-            Some(_) => Ok(true),
+            None => Ok(GetFileFromFileStorageResult::FileNotFound),
+            Some(file_metadata) => {
+                let stored_chunks = read_file_storage
+                    .stored_chunks_count(&file_key)
+                    .map_err(into_rpc_error)?;
+                let total_chunks = file_metadata.chunks_count();
+                if stored_chunks < total_chunks {
```

Comment:
Should we check for an exact match of total chunks as well and return something like `InconsistentFile` (when the number of chunks returned from storage are greater then expected).

Reply:
Good question given the highlighted code, however I'm not sure about codifying the client interface with possible bug scenarios. (Why should I as a client care about you not having a proper file in storage? - just tell me you don't have it and log it for the server operator to see it).

Reply:
Well as a user, I want to know if my data stored is consistent with what I expect it to be. If it isn't then, I would re-upload the data to ensure it is. We are not just saying if the file exists or not, we are also saying if we are missing data. I think we should also cover the opposite **unlikely** event we have more data then what is expected.

Reply:
Changed it to account for this!